File: app/src/main/java/com/katkov/lolachievements/data/local/service/LoginService.java
package com.katkov.lolachievements.data.local.service;

import com.katkov.lolachievements.domain.model.EntryInfoModel;
import com.katkov.lolachievements.data.local.prefser.EntryInfoHolder;

import javax.inject.Inject;

public class LoginService {

    private final EntryInfoHolder entryInfoHolder;

    @Inject
    public LoginService(EntryInfoHolder entryInfoHolder) {
        this.entryInfoHolder = entryInfoHolder;
    }

    public void saveEntryInfo(EntryInfoModel entryInfoModel) {
        entryInfoHolder.putEntryInfo(entryInfoModel);
    }

    public void removeEntryInfo() {
        entryInfoHolder.removeEntryInfo();
    }
}

File: app/src/main/java/com/katkov/lolachievements/data/local/service/LoginService.kt
package com.katkov.lolachievements.data.local.service

import com.katkov.lolachievements.data.local.prefser.LoginModelHolder
import com.katkov.lolachievements.domain.model.LoginModel
import javax.inject.Inject

class LoginService
@Inject
constructor(private val loginModelHolder: LoginModelHolder) {

    fun saveLoginModel(loginModel: LoginModel) {
        loginModelHolder.putLoginModel(loginModel)
    }

    fun removeLoginModel() {
        loginModelHolder.removeLoginModel()
    }
}
